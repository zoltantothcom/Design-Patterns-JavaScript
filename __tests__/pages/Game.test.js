import React from 'react';
import 'jest-styled-components';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { themeLight } from '../../src/styles/themes/theme.light';
import { themeDark } from '../../src/styles/themes/theme.dark';
import styleLight from '../../src/styles/hljs/hljs.light';
import styleDark from '../../src/styles/hljs/hljs.dark';
import Game from '../../src/pages/Game';
import { START, RESTART } from '../../src/static/constants/actions';

const patterns = [
  {
    uuid: 'abc123',
    name: 'Prototype',
    type: 'creational',
    codeES5: 'Code ES5 - Prototype',
    codeES6: 'Code ES6 - Prototype',
    answered: false,
    correct: null,
    variants: [{
      uuid: 'abc234',
      name: 'SIngleton'
    }, {
      uuid: 'abc123',
      name: 'Prototype'
    }]
  },

  {
    uuid: 'abc234',
    name: 'SIngleton',
    type: 'creational',
    codeES5: 'Code ES5 - Singleton',
    codeES6: 'Code ES6 - Singleton',
    answered: false,
    correct: null,
    variants: [{
      uuid: 'abc234',
      name: 'SIngleton'
    }, {
      uuid: 'abc123',
      name: 'Prototype'
    }]
  }
];

const mockStore = configureMockStore();
const store = mockStore({
  patterns: patterns,
  progress: {
    answers: [],
    remaining: [patterns[1]],
    current: patterns[0]
  },
  intro: false,
  mode: 'light',
  js: 'es5'
});

describe('Game page - GAME - LIGHT style', () => {
  let tree;

  beforeEach(() => {
    tree = mount(
      <Provider store={store}>
        <ThemeProvider theme={themeLight}>
          <Game style={styleLight} />
        </ThemeProvider>
      </Provider>
    );
  });

  it('renders a <ProgressBar /> component', () => {
    expect(tree.find('ProgressBar')).toMatchSnapshot();
  });

  it('renders a <Code /> component', () => {
    expect(tree.find('Code')).toMatchSnapshot();
  });

  it('renders the correct code snippet', () => {
    expect(tree.find('Code').text()).toMatch('Code ES5 - Prototype');
  });

  it('renders a <ButtonContainer /> component', () => {
    expect(tree.find('ButtonContainer')).toBeTruthy();
  });

  it('renders <ButtonContainer /> with 2 buttons', () => {
    expect(tree.find('ButtonContainer').find('button')).toHaveLength(2);
  });
});

describe('Game page - GAME - DARK style', () => {
  let tree;

  beforeEach(() => {
    tree = mount(
      <Provider store={store}>
        <ThemeProvider theme={themeDark}>
          <Game style={styleDark} />
        </ThemeProvider>
      </Provider>
    );
  });

  it('renders a <ProgressBar /> component', () => {
    expect(tree.find('ProgressBar')).toMatchSnapshot();
  });

  it('renders a <Code /> component', () => {
    expect(tree.find('Code')).toMatchSnapshot();
  });

  it('renders the correct code snippet', () => {
    expect(tree.find('Code').text()).toMatch('Code ES5 - Prototype');
  });

  it('renders a <ButtonContainer /> component', () => {
    expect(tree.find('ButtonContainer')).toBeTruthy();
  });

  it('renders <ButtonContainer /> with 2 buttons', () => {
    expect(tree.find('ButtonContainer').find('button')).toHaveLength(2);
  });
});

describe('Game page - INTRO', () => {
  let tree;
  const store = mockStore({
    patterns: patterns,
    progress: {
      answers: patterns,
      remaining: [],
      current: null
    },
    intro: true,
    mode: 'light',
    js: 'es5'
  });

  beforeEach(() => {
    tree = mount(
      <Provider store={store}>
        <ThemeProvider theme={themeLight}>
          <Game style={styleLight} />
        </ThemeProvider>
      </Provider>
    );
  });

  it('renders a <Button /> component', () => {
    expect(tree.find('Button')).toMatchSnapshot();
  });

  it('reacts to button click', () => {
    tree.find('button').simulate('click');
    const actions = store.getActions();
    expect(actions).toMatchObject([{ type: START }]);
  });
});

describe('Game page - RESULTS', () => {
  let tree;
  const patterns = [
    {
      uuid: 'abc123',
      name: 'Prototype',
      type: 'creational',
      codeES5: 'Code ES5 - Prototype',
      codeES6: 'Code ES6 - Prototype',
      answered: true,
      correct: true,
      variantUuid: 'abc123',
      variants: [{
        uuid: 'abc234',
        name: 'SIngleton'
      }, {
        uuid: 'abc123',
        name: 'Prototype'
      }]
    },

    {
      uuid: 'abc234',
      name: 'SIngleton',
      type: 'creational',
      codeES5: 'Code ES5 - Singleton',
      codeES6: 'Code ES6 - Singleton',
      answered: true,
      correct: false,
      variantUuid: 'abc123',
      variants: [{
        uuid: 'abc234',
        name: 'SIngleton'
      }, {
        uuid: 'abc123',
        name: 'Prototype'
      }]
    }
  ];
  const store = mockStore({
    patterns: patterns,
    progress: {
      answers: patterns,
      remaining: [],
      current: null
    },
    intro: false,
    mode: 'light',
    js: 'es5'
  });

  beforeEach(() => {
    tree = mount(
      <Provider store={store}>
        <ThemeProvider theme={themeLight}>
          <Game style={styleLight} />
        </ThemeProvider>
      </Provider>
    );
  });

  it('renders a <Result /> component', () => {
    expect(tree.find('Result')).toMatchSnapshot();
  });

  it('renders a <Percentage /> component', () => {
    expect(tree.find('Percentage')).toMatchSnapshot();
  });

  it('renders the correct percentage', () => {
    expect(tree.find('Percentage').text()).toMatch('5%');
  });

  it('renders a <Restart /> component', () => {
    expect(tree.find('ButtonContainer')).toMatchSnapshot();
  });

  it('reacts to button click', () => {
    // button restart and 2 button with incorrect answers
    expect(tree.find('button')).toHaveLength(3);
    // button restart
    tree.find('button#try_again').simulate('click');
    const actions = store.getActions();
    expect(actions).toMatchObject([{ type: RESTART }]);
  });
});
