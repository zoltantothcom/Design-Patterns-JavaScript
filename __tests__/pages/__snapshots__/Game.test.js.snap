// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game page - GAME - DARK style renders a <Code /> component 1`] = `
<Code
  current={
    Object {
      "answered": false,
      "codeES5": "Code ES5 - Prototype",
      "codeES6": "Code ES6 - Prototype",
      "correct": null,
      "name": "Prototype",
      "type": "creational",
      "uuid": "abc123",
    }
  }
  dispatch={[Function]}
  js="es5"
  style={
    Object {
      "hljs": Object {
        "background": "#282828",
        "border": "1px solid #555",
        "color": "#ebdbb2",
        "display": "block",
        "overflowX": "auto",
        "padding": "1em",
      },
      "hljs-addition": Object {
        "color": "#b8bb26",
      },
      "hljs-attr": Object {
        "color": "#fabd2f",
      },
      "hljs-attribute": Object {
        "color": "#8ec07c",
      },
      "hljs-built_in": Object {
        "color": "#83a598",
      },
      "hljs-builtin-name": Object {
        "color": "#8f3f71",
      },
      "hljs-bullet": Object {
        "color": "#8ec07c",
      },
      "hljs-class": Object {
        "color": "#8ec07c",
      },
      "hljs-code": Object {
        "color": "#fe8019",
      },
      "hljs-comment": Object {
        "color": "#928374",
        "fontStyle": "italic",
      },
      "hljs-deletion": Object {
        "color": "#fb4934",
      },
      "hljs-doctag": Object {
        "color": "#8f3f71",
      },
      "hljs-emphasis": Object {
        "color": "#83a598",
        "fontStyle": "italic",
      },
      "hljs-formula": Object {
        "color": "#fb4934",
      },
      "hljs-function": Object {
        "color": "#8ec07c",
      },
      "hljs-function .hljs-keyword": Object {
        "color": "#8ec07c",
      },
      "hljs-keyword": Object {
        "color": "#fb4934",
      },
      "hljs-link": Object {
        "color": "#fb4934",
      },
      "hljs-link_label": Object {
        "color": "#d3869b",
      },
      "hljs-literal": Object {
        "color": "#d3869b",
      },
      "hljs-meta": Object {
        "color": "#fe8019",
      },
      "hljs-meta-keyword": Object {
        "color": "#8ec07c",
      },
      "hljs-meta-string": Object {
        "color": "#b8bb26",
      },
      "hljs-name": Object {
        "color": "#83a598",
      },
      "hljs-number": Object {
        "color": "#d3869b",
      },
      "hljs-params": Object {
        "color": "#fabd2f",
      },
      "hljs-quote": Object {
        "color": "#83a598",
      },
      "hljs-regexp": Object {
        "color": "#fe8019",
      },
      "hljs-section": Object {
        "color": "#b8bb26",
        "fontWeight": "bold",
      },
      "hljs-selector-attr": Object {
        "color": "#b8bb26",
      },
      "hljs-selector-class": Object {
        "color": "#b8bb26",
      },
      "hljs-selector-id": Object {
        "color": "#fe8019",
      },
      "hljs-selector-pseudo": Object {
        "color": "#8ec07c",
      },
      "hljs-selector-tag": Object {
        "color": "#fb4934",
      },
      "hljs-string": Object {
        "color": "#b8bb26",
      },
      "hljs-strong": Object {
        "color": "#83a598",
        "fontWeight": "bold",
      },
      "hljs-subst": Object {
        "color": "#ebdbb2",
      },
      "hljs-symbol": Object {
        "color": "#b8bb26",
      },
      "hljs-tag": Object {
        "color": "#8ec07c",
        "fontWeight": "bold",
      },
      "hljs-template-tag": Object {
        "color": "#fabd2f",
      },
      "hljs-template-variable": Object {
        "color": "#fe8019",
      },
      "hljs-title": Object {
        "color": "#83a598",
      },
      "hljs-type": Object {
        "color": "#fabd2f",
      },
      "hljs-variable": Object {
        "color": "#83a598",
      },
    }
  }
>
  <SyntaxHighlighter
    className="fixed"
    language="javascript"
    style={
      Object {
        "hljs": Object {
          "background": "#282828",
          "border": "1px solid #555",
          "color": "#ebdbb2",
          "display": "block",
          "overflowX": "auto",
          "padding": "1em",
        },
        "hljs-addition": Object {
          "color": "#b8bb26",
        },
        "hljs-attr": Object {
          "color": "#fabd2f",
        },
        "hljs-attribute": Object {
          "color": "#8ec07c",
        },
        "hljs-built_in": Object {
          "color": "#83a598",
        },
        "hljs-builtin-name": Object {
          "color": "#8f3f71",
        },
        "hljs-bullet": Object {
          "color": "#8ec07c",
        },
        "hljs-class": Object {
          "color": "#8ec07c",
        },
        "hljs-code": Object {
          "color": "#fe8019",
        },
        "hljs-comment": Object {
          "color": "#928374",
          "fontStyle": "italic",
        },
        "hljs-deletion": Object {
          "color": "#fb4934",
        },
        "hljs-doctag": Object {
          "color": "#8f3f71",
        },
        "hljs-emphasis": Object {
          "color": "#83a598",
          "fontStyle": "italic",
        },
        "hljs-formula": Object {
          "color": "#fb4934",
        },
        "hljs-function": Object {
          "color": "#8ec07c",
        },
        "hljs-function .hljs-keyword": Object {
          "color": "#8ec07c",
        },
        "hljs-keyword": Object {
          "color": "#fb4934",
        },
        "hljs-link": Object {
          "color": "#fb4934",
        },
        "hljs-link_label": Object {
          "color": "#d3869b",
        },
        "hljs-literal": Object {
          "color": "#d3869b",
        },
        "hljs-meta": Object {
          "color": "#fe8019",
        },
        "hljs-meta-keyword": Object {
          "color": "#8ec07c",
        },
        "hljs-meta-string": Object {
          "color": "#b8bb26",
        },
        "hljs-name": Object {
          "color": "#83a598",
        },
        "hljs-number": Object {
          "color": "#d3869b",
        },
        "hljs-params": Object {
          "color": "#fabd2f",
        },
        "hljs-quote": Object {
          "color": "#83a598",
        },
        "hljs-regexp": Object {
          "color": "#fe8019",
        },
        "hljs-section": Object {
          "color": "#b8bb26",
          "fontWeight": "bold",
        },
        "hljs-selector-attr": Object {
          "color": "#b8bb26",
        },
        "hljs-selector-class": Object {
          "color": "#b8bb26",
        },
        "hljs-selector-id": Object {
          "color": "#fe8019",
        },
        "hljs-selector-pseudo": Object {
          "color": "#8ec07c",
        },
        "hljs-selector-tag": Object {
          "color": "#fb4934",
        },
        "hljs-string": Object {
          "color": "#b8bb26",
        },
        "hljs-strong": Object {
          "color": "#83a598",
          "fontWeight": "bold",
        },
        "hljs-subst": Object {
          "color": "#ebdbb2",
        },
        "hljs-symbol": Object {
          "color": "#b8bb26",
        },
        "hljs-tag": Object {
          "color": "#8ec07c",
          "fontWeight": "bold",
        },
        "hljs-template-tag": Object {
          "color": "#fabd2f",
        },
        "hljs-template-variable": Object {
          "color": "#fe8019",
        },
        "hljs-title": Object {
          "color": "#83a598",
        },
        "hljs-type": Object {
          "color": "#fabd2f",
        },
        "hljs-variable": Object {
          "color": "#83a598",
        },
      }
    }
  >
    <pre
      className="fixed"
      style={
        Object {
          "background": "#282828",
          "border": "1px solid #555",
          "color": "#ebdbb2",
          "display": "block",
          "overflowX": "auto",
          "padding": "1em",
        }
      }
    >
      <code>
        Code ES5 - Prototype
      </code>
    </pre>
  </SyntaxHighlighter>
</Code>
`;

exports[`Game page - GAME - DARK style renders a <ProgressBar /> component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 3px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.25rem 0 1rem;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2px;
  width: 3.75%;
}

<ProgressBar
  answers={Array []}
  dispatch={[Function]}
  remaining={
    Array [
      Object {
        "answered": false,
        "codeES5": "Code ES5 - Singleton",
        "codeES6": "Code ES6 - Singleton",
        "correct": null,
        "name": "SIngleton",
        "type": "creational",
        "uuid": "abc234",
      },
    ]
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  height: 3px;
  justify-content: space-between;
  margin: 1.25rem 0 1rem;
  width: 100%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <styled.span
          key="abc234"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  background: ",
                    [Function],
                    ";
  display: flex;
  height: 2px;
  width: 3.75%;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c1"
            />
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</ProgressBar>
`;

exports[`Game page - GAME - LIGHT style renders a <Code /> component 1`] = `
<Code
  current={
    Object {
      "answered": false,
      "codeES5": "Code ES5 - Prototype",
      "codeES6": "Code ES6 - Prototype",
      "correct": null,
      "name": "Prototype",
      "type": "creational",
      "uuid": "abc123",
    }
  }
  dispatch={[Function]}
  js="es5"
  style={
    Object {
      "hljs": Object {
        "background": "#fafafa",
        "border": "1px solid #d8d8d8",
        "color": "#383a42",
        "display": "block",
        "overflowX": "auto",
        "padding": "1em",
      },
      "hljs-addition": Object {
        "color": "#50a14f",
      },
      "hljs-attr": Object {
        "color": "#986801",
      },
      "hljs-attribute": Object {
        "color": "#50a14f",
      },
      "hljs-built_in": Object {
        "color": "#c18401",
      },
      "hljs-bullet": Object {
        "color": "#4078f2",
      },
      "hljs-class .hljs-title": Object {
        "color": "#c18401",
      },
      "hljs-comment": Object {
        "color": "#a0a1a7",
        "fontStyle": "italic",
      },
      "hljs-deletion": Object {
        "color": "#e45649",
      },
      "hljs-doctag": Object {
        "color": "#a626a4",
      },
      "hljs-emphasis": Object {
        "fontStyle": "italic",
      },
      "hljs-formula": Object {
        "color": "#a626a4",
      },
      "hljs-keyword": Object {
        "color": "#a626a4",
      },
      "hljs-link": Object {
        "color": "#4078f2",
        "textDecoration": "underline",
      },
      "hljs-literal": Object {
        "color": "#0184bb",
      },
      "hljs-meta": Object {
        "color": "#4078f2",
      },
      "hljs-meta-string": Object {
        "color": "#50a14f",
      },
      "hljs-name": Object {
        "color": "#e45649",
      },
      "hljs-number": Object {
        "color": "#986801",
      },
      "hljs-quote": Object {
        "color": "#a0a1a7",
        "fontStyle": "italic",
      },
      "hljs-regexp": Object {
        "color": "#50a14f",
      },
      "hljs-section": Object {
        "color": "#e45649",
      },
      "hljs-selector-attr": Object {
        "color": "#986801",
      },
      "hljs-selector-class": Object {
        "color": "#986801",
      },
      "hljs-selector-id": Object {
        "color": "#4078f2",
      },
      "hljs-selector-pseudo": Object {
        "color": "#986801",
      },
      "hljs-selector-tag": Object {
        "color": "#e45649",
      },
      "hljs-string": Object {
        "color": "#50a14f",
      },
      "hljs-strong": Object {
        "fontWeight": "bold",
      },
      "hljs-subst": Object {
        "color": "#e45649",
      },
      "hljs-symbol": Object {
        "color": "#4078f2",
      },
      "hljs-template-variable": Object {
        "color": "#986801",
      },
      "hljs-title": Object {
        "color": "#4078f2",
      },
      "hljs-type": Object {
        "color": "#986801",
      },
      "hljs-variable": Object {
        "color": "#986801",
      },
    }
  }
>
  <SyntaxHighlighter
    className="fixed"
    language="javascript"
    style={
      Object {
        "hljs": Object {
          "background": "#fafafa",
          "border": "1px solid #d8d8d8",
          "color": "#383a42",
          "display": "block",
          "overflowX": "auto",
          "padding": "1em",
        },
        "hljs-addition": Object {
          "color": "#50a14f",
        },
        "hljs-attr": Object {
          "color": "#986801",
        },
        "hljs-attribute": Object {
          "color": "#50a14f",
        },
        "hljs-built_in": Object {
          "color": "#c18401",
        },
        "hljs-bullet": Object {
          "color": "#4078f2",
        },
        "hljs-class .hljs-title": Object {
          "color": "#c18401",
        },
        "hljs-comment": Object {
          "color": "#a0a1a7",
          "fontStyle": "italic",
        },
        "hljs-deletion": Object {
          "color": "#e45649",
        },
        "hljs-doctag": Object {
          "color": "#a626a4",
        },
        "hljs-emphasis": Object {
          "fontStyle": "italic",
        },
        "hljs-formula": Object {
          "color": "#a626a4",
        },
        "hljs-keyword": Object {
          "color": "#a626a4",
        },
        "hljs-link": Object {
          "color": "#4078f2",
          "textDecoration": "underline",
        },
        "hljs-literal": Object {
          "color": "#0184bb",
        },
        "hljs-meta": Object {
          "color": "#4078f2",
        },
        "hljs-meta-string": Object {
          "color": "#50a14f",
        },
        "hljs-name": Object {
          "color": "#e45649",
        },
        "hljs-number": Object {
          "color": "#986801",
        },
        "hljs-quote": Object {
          "color": "#a0a1a7",
          "fontStyle": "italic",
        },
        "hljs-regexp": Object {
          "color": "#50a14f",
        },
        "hljs-section": Object {
          "color": "#e45649",
        },
        "hljs-selector-attr": Object {
          "color": "#986801",
        },
        "hljs-selector-class": Object {
          "color": "#986801",
        },
        "hljs-selector-id": Object {
          "color": "#4078f2",
        },
        "hljs-selector-pseudo": Object {
          "color": "#986801",
        },
        "hljs-selector-tag": Object {
          "color": "#e45649",
        },
        "hljs-string": Object {
          "color": "#50a14f",
        },
        "hljs-strong": Object {
          "fontWeight": "bold",
        },
        "hljs-subst": Object {
          "color": "#e45649",
        },
        "hljs-symbol": Object {
          "color": "#4078f2",
        },
        "hljs-template-variable": Object {
          "color": "#986801",
        },
        "hljs-title": Object {
          "color": "#4078f2",
        },
        "hljs-type": Object {
          "color": "#986801",
        },
        "hljs-variable": Object {
          "color": "#986801",
        },
      }
    }
  >
    <pre
      className="fixed"
      style={
        Object {
          "background": "#fafafa",
          "border": "1px solid #d8d8d8",
          "color": "#383a42",
          "display": "block",
          "overflowX": "auto",
          "padding": "1em",
        }
      }
    >
      <code>
        Code ES5 - Prototype
      </code>
    </pre>
  </SyntaxHighlighter>
</Code>
`;

exports[`Game page - GAME - LIGHT style renders a <ProgressBar /> component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 3px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.25rem 0 1rem;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2px;
  width: 3.75%;
}

<ProgressBar
  answers={Array []}
  dispatch={[Function]}
  remaining={
    Array [
      Object {
        "answered": false,
        "codeES5": "Code ES5 - Singleton",
        "codeES6": "Code ES6 - Singleton",
        "correct": null,
        "name": "SIngleton",
        "type": "creational",
        "uuid": "abc234",
      },
    ]
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  height: 3px;
  justify-content: space-between;
  margin: 1.25rem 0 1rem;
  width: 100%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <styled.span
          key="abc234"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  background: ",
                    [Function],
                    ";
  display: flex;
  height: 2px;
  width: 3.75%;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c1"
            />
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</ProgressBar>
`;

exports[`Game page - INTRO renders a <Button /> component 1`] = `
.c0 {
  background-color: #F2E8F2;
  border: 1px solid #EDB8ED;
  border-radius: 4px;
  cursor: pointer;
  font: 400 1rem 'Karla','sans-serif';
  height: 48px;
  max-width: 240px;
  outline: none;
  padding: 0 1.5rem;
  width: 37.5%;
}

.c0 span {
  color: #6F256F;
}

.c0:hover {
  background-color: #6F256F;
  border-color: #6F256F;
}

.c0:hover span {
  color: #FFFFFF;
}

<Button
  id="start"
  label="Start Game"
  onClick={[Function]}
>
  <styled.button
    id="start"
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  background-color: ",
              [Function],
              ";
  border: 1px solid ",
              [Function],
              ";
  border-radius: 4px;
  cursor: pointer;
  font: 400 1rem 'Karla', 'sans-serif';
  height: 48px;
  max-width: 240px;
  outline: none;
  padding: 0 1.5rem;
  width: 37.5%;

  & span {
    color: ",
              [Function],
              ";
  }

  &:hover {
    background-color: ",
              [Function],
              ";
    border-color: ",
              [Function],
              ";

    & span {
      color: ",
              [Function],
              ";
    }
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      id="start"
      onClick={[Function]}
    >
      <button
        className="c0"
        id="start"
        onClick={[Function]}
      >
        <span>
          Start Game
        </span>
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;

exports[`Game page - RESULTS renders a <Percentage /> component 1`] = `
.c0 {
  border: 4px solid red;
  border-radius: 50%;
  color: red;
  font-family: 'Karla','sans-serif';
  font-size: 4rem;
  height: 10rem;
  margin: auto;
  line-height: 10rem;
  text-align: center;
  width: 10rem;
}

<Percentage
  answers={
    Array [
      Object {
        "answered": true,
        "codeES5": "Code ES5 - Prototype",
        "codeES6": "Code ES6 - Prototype",
        "correct": true,
        "name": "Prototype",
        "type": "creational",
        "uuid": "abc123",
      },
      Object {
        "answered": true,
        "codeES5": "Code ES5 - Singleton",
        "codeES6": "Code ES6 - Singleton",
        "correct": false,
        "name": "SIngleton",
        "type": "creational",
        "uuid": "abc234",
      },
    ]
  }
  dispatch={[Function]}
>
  <styled.h1
    level="red"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  border: 4px solid ",
              [Function],
              ";
  border-radius: 50%;
  color: ",
              [Function],
              ";
  font-family: 'Karla', 'sans-serif';
  font-size: 4rem;
  height: 10rem;
  margin: auto;
  line-height: 10rem;
  text-align: center;
  width: 10rem;
",
            ],
          },
          "displayName": "styled.h1",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "h1",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      level="red"
    >
      <h1
        className="c0"
      >
        5
        %
      </h1>
    </StyledComponent>
  </styled.h1>
</Percentage>
`;

exports[`Game page - RESULTS renders a <Restart /> component 1`] = `null`;

exports[`Game page - RESULTS renders a <Result /> component 1`] = `
.c0 {
  color: #6F256F;
  margin: 3rem 0;
  text-align: center;
}

<Result
  answers={
    Array [
      Object {
        "answered": true,
        "codeES5": "Code ES5 - Prototype",
        "codeES6": "Code ES6 - Prototype",
        "correct": true,
        "name": "Prototype",
        "type": "creational",
        "uuid": "abc123",
      },
      Object {
        "answered": true,
        "codeES5": "Code ES5 - Singleton",
        "codeES6": "Code ES6 - Singleton",
        "correct": false,
        "name": "SIngleton",
        "type": "creational",
        "uuid": "abc234",
      },
    ]
  }
  dispatch={[Function]}
>
  <styled.p>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  margin: 3rem 0;
  text-align: center;
",
            ],
          },
          "displayName": "styled.p",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "p",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <p
        className="c0"
      >
        You got 
        <strong>
          1
        </strong>
         patterns right out of 
        2
        .
      </p>
    </StyledComponent>
  </styled.p>
</Result>
`;
